pkg/actions/lua/lakefs/catalogexport/symlink_exporter.lua

```diff
@@ -0,0 +1,93 @@
+local extractor = require("lakefs/catalogexport/table_extractor")
+local hive = require("lakefs/catalogexport/hive")
+local utils = require("lakefs/catalogexport/internal")
+local url = require("net/url")
+local pathlib = require("path")
+local strings = require("strings")
+
+-- make it a function so that @nopcoder can ask why we bloat the code? 
+local function parse_storage_uri(uri)
```

Comment:
yes, you are bloating the code - please don't get table and assign each field to local just to return them. if you are not doing anything with a field just use the member, I don't need to meet another local and remember what it does.

also, you can see that the writter of url.parse already return a struct/table, and the beauty of the members describe each field - the current function split them into local variables in which we loose this marbel.

from the go code path can't be nil and we need to match it with the separator before we drop it.

suggest the following code + internal.lua
```lua
local function parse_storage_uri(uri)
    local u = url.parse(uri)
    return { 
        protocol = u.scheme,
        bucket = u.host,
        key = (u.path:sub(0, 1) == "/") and u.path:sub(2) or u.path,
    }
end
```

Reply:
💯